src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1439,574 +1687,35 @@ internal void Stop()
         /// </summary>
         internal void Done()
         {
-            if (_inputThread != null)
+            // we allow call Done() multiply times.
+            // For example one time from Process() code path, 
+            // when we detect that process already finished
+            // and once from End() code path.
```

Comment:
`Process()` and `End()` is kinda confusing at first glance. Can you please change `Process()` to `ProcessRecord()` and `End()` to `Complete()`?
And please add one more comment explains that "even though `Done` could be called multiple times, the calls are on the same thread, so there is no race condition here".